src/wire/dhcp.rs

```diff
@@ -0,0 +1,629 @@
+    pub const OPT_DOMAIN_NAME: u8 = 15;
+    pub const OPT_SWAP_SERVER: u8 = 16;
+    pub const OPT_ROOT_PATH: u8 = 17;
+    pub const OPT_EXTENSIONS_PATH: u8 = 18;
+
+    // IP Layer Parameters per Host
+    pub const OPT_IP_FORWARDING: u8 = 19;
+    pub const OPT_NON_LOCAL_SOURCE_ROUTING: u8 = 20;
+    pub const OPT_POLICY_FILTER: u8 = 21;
+    pub const OPT_MAX_DATAGRAM_REASSEMBLY_SIZE: u8 = 22;
+    pub const OPT_DEFAULT_TTL: u8 = 23;
+    pub const OPT_PATH_MTU_AGING_TIMEOUT: u8 = 24;
+    pub const OPT_PATH_MTU_PLATEU_TABLE: u8 = 25;
+
+    // IP Layer Parameters per Interface
+    pub const OPT_INTERFACE_MTU: u8 = 26;
+    pub const OPT_ALL_SUBNETS_ARE_LOCAL: u8 = 27;
+    pub const OPT_BROADCAST_ADDRESS: u8 = 28;
+    pub const OPT_PERFORM_MASK_DISCOVERY: u8 = 29;
+    pub const OPT_MASK_SUPPLIER: u8 = 30;
+    pub const OPT_PERFORM_ROUTER_DISCOVERY: u8 = 31;
+    pub const OPT_ROUTER_SOLICITATION_ADDRESS: u8 = 32;
+    pub const OPT_STATIC_ROUTE: u8 = 33;
+
+    // Link Layer Parameters per Interface
+    pub const OPT_TRAILER_ENCAPSULATION: u8 = 34;
+    pub const OPT_ARP_CACHE_TIMEOUT: u8 = 35;
+    pub const OPT_ETHERNET_ENCAPSULATION: u8 = 36;
+
+    // TCP Parameters
+    pub const OPT_TCP_DEFAULT_TTL: u8 = 37;
+    pub const OPT_TCP_KEEPALIVE_INTERVAL: u8 = 38;
+    pub const OPT_TCP_KEEPALIVE_GARBAGE: u8 = 39;
+
+    // Application and Service Parameters
+    pub const OPT_NIS_DOMAIN: u8 = 40;
+    pub const OPT_NIS_SERVERS: u8 = 41;
+    pub const OPT_NTP_SERVERS: u8 = 42;
+    pub const OPT_VENDOR_SPECIFIC_INFO: u8 = 44;
+    pub const OPT_NETBIOS_SERVER: u8 = 45;
+    pub const OPT_NETBIOS_NODE_TYPE: u8 = 46;
+    pub const OPT_NETBIOS_SCOPE: u8 = 47;
+    pub const OPT_X_WINDOW_FONT_SERVER: u8 = 48;
+    pub const OPT_X_WINDOW_DISPLAY_MANAGER: u8 = 49;
+    pub const OPT_NIS_PLUS_DOMAIN: u8 = 64;
+    pub const OPT_NIS_PLUS_SERVERS: u8 = 65;
+    pub const OPT_MOBILE_IP_HOME_AGENT: u8 = 68;
+    pub const OPT_SMTP_SERVER: u8 = 69;
+    pub const OPT_POP3_SERVER: u8 = 70;
+    pub const OPT_NNTP_SERVER: u8 = 71;
+    pub const OPT_WWW_SERVER: u8 = 72;
+    pub const OPT_FINGER_SERVER: u8 = 73;
+    pub const OPT_IRC_SERVER: u8 = 74;
+    pub const OPT_STREETTALK_SERVER: u8 = 75;
+    pub const OPT_STDA_SERVER: u8 = 76;
+
+    // DHCP Extensions
+    pub const OPT_REQUESTED_IP: u8 = 50;
+    pub const OPT_IP_LEASE_TIME: u8 = 51;
+    pub const OPT_OPTION_OVERLOAD: u8 = 52;
+    pub const OPT_TFTP_SERVER_NAME: u8 = 66;
+    pub const OPT_BOOTFILE_NAME: u8 = 67;
+    pub const OPT_DHCP_MESSAGE_TYPE: u8 = 53;
+    pub const OPT_SERVER_IDENTIFIER: u8 = 54;
+    pub const OPT_PARAMETER_REQUEST_LIST: u8 = 55;
+    pub const OPT_MESSAGE: u8 = 56;
+    pub const OPT_MAX_DHCP_MESSAGE_SIZE: u8 = 57;
+    pub const OPT_RENEWAL_TIME_VALUE: u8 = 58;
+    pub const OPT_REBINDING_TIME_VALUE: u8 = 59;
+    pub const OPT_VENDOR_CLASS_ID: u8 = 60;
+    pub const OPT_CLIENT_ID: u8 = 61;
+}
+
+impl<T: AsRef<[u8]>> Packet<T> {
```

Comment:
Most the other `Packet` types document the methods. The `Repr` is documented well, but it would be nice if this was as well.